internal/protoc/downloader.go

```diff
@@ -64,7 +71,25 @@ func newDownloader(config settings.Config, options ...DownloaderOption) *downloa
 	if downloader.config.Compile.ProtobufVersion == "" {
 		downloader.config.Compile.ProtobufVersion = vars.DefaultProtocVersion
 	}
-	return downloader
+	if downloader.protocBinPath != "" || downloader.protocWktPath != "" {
+		if downloader.protocURL != "" {
+			return nil, fmt.Errorf("cannot use protoc-url in combination with either protoc-bin-path or protoc-wkt-path")
+		}
+		if downloader.protocBinPath == "" || downloader.protocWktPath == "" {
+			return nil, fmt.Errorf("both protoc-bin-path and protoc-wkt-path must be set")
+		}
+		cleanBinPath := filepath.Clean(downloader.protocBinPath)
+		if _, err := os.Stat(cleanBinPath); os.IsNotExist(err) {
+			return nil, err
+		}
+		cleanWktPath := filepath.Clean(downloader.protocWktPath)
+		if _, err := os.Stat(cleanWktPath); os.IsNotExist(err) {
+			return nil, err
+		}
+		downloader.protocBinPath = cleanBinPath
+		downloader.protocWktPath = cleanWktPath
+	}
+	return downloader, nil
```

Comment:
Prototool probably has other things that return errors from constructors, but it is generally better practice to do all these checks within the functions, as is done now for other parts of the Downloader, and not updating the constructor to return error

Reply:
In this case, I'd prefer this to be determined up-front. This way, any repeated calls to `ProtocPath` and/or `WellKnownTypesIncludePath` don't need to re-validate. What do you think?

Reply:
I don’t really care that much either way, this is fine if it works